docs/glossary.rst

```diff
@@ -0,0 +1,68 @@
+.. currentmodule:: plasmapy
+
+********
+Glossary
+********
+
+.. glossary::
+
+   args
+      An abbreviation for `positional arguments`_.
+
+   -like
+      Used to indicate an `object` of that type or that can instantiate
+      that type.  For example, ``"He 2+"`` is :term:`particle-like`
+      because it can be used to instantiate |Particle|.
+
+   charge number
+      The charge of a particle in units of elementary charge.
+
+   keyword-only
+      An argument that must be provided along with the name of that
+      argument. If ``z`` is a keyword-only argument to ``f(z)``, then
+      it can be provided as ``f(z=2)`` but not ``f(2)``.
+
+   kwargs
+      An abbreviation for `keyword arguments`_.
+
+   particle-like
+      An `object` is *particle-like* if it is a |Particle| or
+      |CustomParticle|, or can be cast into one.
```

Comment:
So it's just an instance of an abstract particle or a string that cleanly initializes one. I did just wake up, haven't had tea yet, excuse my confusion!

Reply:
Pretty much!  It could also be an `int` representing an atomic number.  I also added the line from the docstring for `ParticleLike` that describes why a `DimesionlessParticle` isn't particle-like.